satellite/satellitedb/overlaycache.go

```diff
@@ -299,24 +299,21 @@ func (cache *overlaycache) postgresQueryNodesDistinct(ctx context.Context, exclu
 	args = append(args, count)
 
 	rows, err := cache.db.Query(cache.db.Rebind(`
-	WITH candidates AS (
-		SELECT * FROM nodes
-		`+safeQuery+safeExcludeNodes+safeExcludeIPs+`
-	)
-	SELECT
-		id, type, address, last_net, free_bandwidth, free_disk, total_audit_count,
+	SELECT *
+	FROM (
+		SELECT DISTINCT ON (last_net) last_net,    -- choose at max 1 node from this IP or network
```

Comment:
This is reordering the output columns, I've just noticed, and the code below that scans the results into local variables is not changed in the same way, so it will no longer work (or if it does, it will be entirely wrong).

Reply:
Great review, I missed that. Scans are reordered now.